scratch/xbgpu/benchmarks/correlate_bench.py

```diff
@@ -52,7 +53,12 @@ def main():
     fn = template.instantiate(command_queue, args.heaps_per_fengine_per_chunk)
 
     fn.ensure_all_bound()
-    fn.buffer("in_samples").zero(command_queue)
+    buf = fn.buffer("in_samples")
+    h_data = buf.empty_like()
+    assert h_data.dtype == np.int8
+    rng = np.random.default_rng(seed=1)
+    h_data = rng.integers(-128, 127, size=h_data.shape, dtype=h_data.dtype)
```

Comment:
You're creating h_data, then discarding it to be garbage collected and replacing it with a different array. You could simplify things like so:
```suggestion
    assert buf.dtype == np.int8
    rng = np.random.default_rng(seed=1)
    h_data = rng.integers(-128, 127, size=h_data.shape, dtype=buf.dtype)
```

Reply:
Fair enough. I gather you meant the `size` arg to be `buf.shape`. Will update now-now.

Reply:
That would be fine too. Since it's done elsewhere let's go with that.